### PR TITLE
added detect contour

### PR DESCRIPTION
**Pull request by lplp8899**
Added two files related to contour detection. More files come later.

### CODE REVIEW

**Comment by kirkscheper** [comment]
Can you give some explanation of what this should do?

**Comment by flixr** [comment]
Seems there still quite a few things missing...
Also please correct the file name in the doxygen header and remove the old commented functions...

**Comment by rmeertens** [comment]
Looks like the file "modules/computer_vision/opencv_contour.h" is missing :(

**Comment by lplp8899** [comment]
add the opencv_contour.cpp and .h files, hope there is no problem.

File: sw/airborne/modules/computer_vision/opencv_contour.cpp [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+  ///*
+  int y1, y2, u1, u2, v1, v2;
+  
+
+  y1=16;  u1=135; v1=80; // in the mavlab, bright
+  y2=100; u2=175; v2=165;
+  // for cyberzoo: Y:12-95, U:129-161, V:80-165, turn white.     
+  //int y1=16;  int u1=129; int v1=80; % cyberzoo, dark
+  //int y2=100; int u2=161; int v2=165;
+
```
Comment (kirkscheper) [inline sw/airborne/modules/computer_vision/opencv_contour.cpp]: What are these values for? Why hardcoded? how do you select them?

File: sw/airborne/modules/computer_vision/opencv_contour.cpp [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+  Mat image, edge_image, thresh_image;
+
+  // 
+  //blur(image, image, Size(5, 5));
+  //GaussianBlur(image, image, Size(9,9), 5,5);
+
+  // Canny edges, only works with grayscale image
+  //int edgeThresh = 35;
+  //Canny(image, image, edgeThresh, edgeThresh * 3);
+
```
Comment (kirkscheper) [inline sw/airborne/modules/computer_vision/opencv_contour.cpp]: There is lots of commented out code, is this a work in progress or is it completed?

File: sw/airborne/modules/computer_vision/opencv_contour.cpp [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+   float Im_center_h = height/2.;
+   
+   float real_size = 1.; // real size of the object
+   d_y = -(rect_center.x - Im_center_w) * real_size/float(bounding_rect.width); // right hand
+   d_z = -(rect_center.y - Im_center_h) * real_size/float(bounding_rect.height); // point downwards
+   
+   
+   printf("dy =%f\n", d_y );  
+   printf("dz =%f\n", d_z );    
+
```
Comment (kirkscheper) [inline sw/airborne/modules/computer_vision/opencv_contour.cpp]: remove printfs

File: sw/airborne/modules/computer_vision/opencv_contour_RM.cpp [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+// img[(row * width + col) * 2] = 0.877 * (pixelHere[0] - img[(row * width + col) * 2 + 1] + 127);
+// }
+// }
+// }
+// }
+//
+convertBackFromGrayscale( gray, img, width, height);
+//convertBackFromRGB( image, img, width, height);
+return 0;
+}
```
Comment (kirkscheper) [inline sw/airborne/modules/computer_vision/opencv_contour_RM.cpp]: there are no tabs in this file, please fun ./fix_code_style on all the files you commit

File: sw/airborne/modules/computer_vision/opencv_contour_RM.h [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+int find_contour(char *img, int width, int height);
+
+//void yuv_opencv_to_yuv422(CV::Mat image, char *img, int width, int height);
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif
```
Comment (kirkscheper) [inline sw/airborne/modules/computer_vision/opencv_contour_RM.h]: tabs

File: sw/airborne/modules/computer_vision/opencv_contour_RM.cpp [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+*
+* You should have received a copy of the GNU General Public License
+* along with paparazzi; see the file COPYING. If not, see
+* <http://www.gnu.org/licenses/>.
+*/
+/**
+* @file "modules/computer_vision/opencv_example.cpp"
+* @author C. De Wagter
+* opencv
+*/
```
Comment (kirkscheper) [inline sw/airborne/modules/computer_vision/opencv_contour_RM.cpp]: Wrong file name and author information

File: sw/airborne/modules/computer_vision/opencv_contour_RM.h [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+ *
+ */
+
+/**
+ * @file modules/computer_vision/opencv_contour.h
+ *
+ * A small library with functions to convert between the Paparazzi used YUV422 arrays
+ * and the opencv image functions.
+ */
+
```
Comment (kirkscheper) [inline sw/airborne/modules/computer_vision/opencv_contour_RM.h]: Wrong file name and author information

File: sw/airborne/modules/computer_vision/detect_contour.c [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+  return FALSE;
+}
+
+void opencvdemo_init(void)
+{
+  //cv_add(opencv_func);
+  cv_add_to_device(&OPENCVDEMO_CAMERA, opencv_func);
+}
+
```
Comment (kirkscheper) [inline sw/airborne/modules/computer_vision/detect_contour.c]: It seems like the file only adds the find_contour to the vision processing thread. Is there not a better way to do this then adding an entire file for this? It also looks like the functions here don't return anything and you have to read the variables from the openncv_contour.h functions and variables, can't you somehow do what you do here in the opencv_contour as well?

Comment (rmeertens) [inline sw/airborne/modules/computer_vision/detect_contour.c]: Mmmm, can you make c++ modules?

Comment (flixr) [inline sw/airborne/modules/computer_vision/detect_contour.c]: sure

**Comment by flixr** [comment]
closed with #1798

**Pull request closed without merging**